### ISSUE TITLE
Various issues in derivatives.yaml

### ISSUE THREAD

**Issue by albanD** [pytorch/pytorch#54591]
While browsing derivatives.yaml to implement forward AD, I found quite a few surprising things there.
We should look into these in detail and update/fix them:

  - [ ] remove link to Functions.cpp in derivatives.yaml
  - [ ] cummin dedup + extra arg
  - [ ] zeros_like that should not be there according to comment!
  - [ ] distributions gradients?
  - [ ] hypot  doc?
  - [ ] matrix_exp doc?
  - [ ] norm(p=inf) with equal values fw and bw mode
  - [ ] renorm backward uses detach??
  - [ ] trilinear_backward unroll_dim
  - [ ] check_no_requires_grad should not run in NoGrad mode
  - [ ] index.Tensor indices differentiability
  - [ ] native_layer_norm contiguous handling
  - [ ] doc for backward should say vector jacobian
